tests/test_scenarios.py

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env python
+import logging
+from pathlib import Path
+
+import pytest
+from conftest import TEST_Path
+from pyrdfj2 import J2RDFSyntaxBuilder
+from util4tests import run_single_test
+
+from travharv import TravHarv
+
+log = logging.getLogger(__name__)
+
+QUERY_BUILDER: J2RDFSyntaxBuilder = J2RDFSyntaxBuilder(
+    templates_folder=str(Path(__file__).parent / "scenarios" / "templates")
+)
+
+BASE = "http://localhost:8080/"
+OUTPUTS = TEST_Path / "output"
+INPUT = TEST_Path / "input"
+CONFIGS = TEST_Path / "travharv_config"
+RDF_MIMES = {
+    "text/turtle",
+    "application/ld+json",
+}
+
+# TODO write out all the assertions for the tests
+# this being the number of triples
+# the documents that ware asserted
+SCENARIOS_OUTCOMES = {
+    "dereference_test1_sparql.yml": [
+        {"expected_len_triples": "19", "mode": "stop"},
```

Comment:
where is difference between these modes is documented and why do they exist?

Reply:
We discussed this in one of our last collab sessions , the no-stop method retries for all subjects until it ensures all property paths given by the user are in the store. The stop method just takes the first subject , tries and if it fails it just continues to the next property âth without retrying for other subjects.